src/box/lua/config/descriptions.lua

```diff
@@ -0,0 +1,2446 @@
+    The degree of parallelism for parallel rebalancing.
+]])
+
+M['sharding.rebalancer_mode'] = format_text([[
+    Configure how a rebalancer is selected:
+
+    -`auto` (default): if there are no replica sets with the rebalancer
+    sharding role (`sharding.roles`), a replica set with the rebalancer is
+    selected automatically among all replica sets.
+    - `manual`: one of the replica sets should have the rebalancer sharding
+    role. The rebalancer is in this replica set.
+    - `off`: rebalancing is turned off regardless of whether a replica set
+    with the rebalancer sharding role exists or not.
+]])
+
+M['sharding.roles'] = format_text([[
+    Roles of a replica set in regard to sharding. A replica set can have the
+    following roles:
+
+    - `router`: a replica set acts as a router.
+    - `storage`: a replica set acts as a storage.
+    - `rebalancer`: a replica set acts as a rebalancer.
+
+    The rebalancer role is optional. If it is not specified, a rebalancer is
+    selected automatically from the master instances of replica sets.
+
+    There can be at most one replica set with the rebalancer role.
+    Additionally, this replica set should have a `storage` role.
+]])
+
+M['sharding.roles.*'] = 'Role name.'
+
+M['sharding.sched_move_quota'] = format_text([[
+    A scheduler's bucket move quota used by the rebalancer.
+
+    `sched_move_quota` defines how many bucket moves can be done in a row if
+    there are pending storage refs. Then, bucket moves are blocked and a router
+    continues making map-reduce requests.
+]])
+
+M['sharding.sched_ref_quota'] = format_text([[
+    A scheduler's storage ref quota used by a router's map-reduce API.
+    For example, the `vshard.router.map_callrw()` function implements
+    consistent map-reduce over the entire cluster.
+
+    `sched_ref_quota` defines how many storage refs, therefore map-reduce
+    requests, can be executed on the storage in a row if there are pending
+    bucket moves. Then, storage refs are blocked and the rebalancer continues
+    bucket moves.
+]])
+
+M['sharding.shard_index'] = format_text([[
+    The name or ID of a TREE index over the bucket id. Spaces without this
+    index do not participate in a sharded Tarantool cluster and can be used
+    as regular spaces if needed. It is necessary to specify the first part of
+    the index, other parts are optional.
+]])
+
+M['sharding.sync_timeout'] = format_text([[
+    The timeout to wait for synchronization of the old master with replicas
+    before demotion. Used when switching a master or when manually calling
+    the `sync()` function.
+]])
+
+M['sharding.weight'] = format_text([[
+    The relative amount of data that a replica set can store.
+]])
+
+M['sharding.zone'] = format_text([[
+    A zone that can be set for routers and replicas. This allows sending
+    read-only requests not only to a master instance but to any available
+    replica that is the nearest to the router.
+]])
```

Comment:
To be honest, this description sounds unclear for me.

Are read-only requests sent only to the master if zones are not set?

Are requests sent to the nearest zone only or are they fall back to other zones if the nearest one is not available?

Are requests sent only to the master if it is in the nearest zone?

Feel free to refuse to collect the details. But if you're interested, this description needs some work, IMHO. If you're tired, let's left it as is -- we can return here back later.

Reply:
I tried searching, but I haven't come up with anything clear yet. Let's return here back later.

Reply:
OK. If it originates from the documentation on the website, I would like if you file issue against the documentation. Closing the discussion here.